### PR TITLE
Build Serac with PETSc

### LINKED ISSUE #1070: Build with PETSC
I'm not sure if we're using PETSC but I noticed this comment about it not being imported. Do we want to update this?

```cmake
  # Petsc
  if(SERAC_USE_PETSC)
    # Note: Targets not currently imported
  endif()
```

https://github.com/LLNL/serac/blob/develop/cmake/serac-config.cmake.in#L139

### LINKED ISSUE #1082: Codeveloping MFEM with PETSc
Since Serac has some logic for finding petsc in it's `SetupSeracThirdParty.cmake`, it creates some problems when codeveloping Serac in LiDO. It has to do with MFEM's `FindPETSc.cmake` module. I need to look into this more once I start working on building PETSc for Serac.

```cmake
    if(PETSC_DIR)
        serac_assert_is_directory(VARIABLE_NAME PETSC_DIR)
        include(${CMAKE_CURRENT_LIST_DIR}/FindPETSc.cmake)
        message(STATUS "PETSc support is ON")
        set(PETSC_FOUND TRUE)
    else()
        message(STATUS "PETSc support is OFF")
        set(PETSC_FOUND FALSE)
    endif()
```

```
CMake Error at smith/serac/mfem/config/cmake/modules/FindPETSc.cmake:147 (message):
  The pair
  PETSC_DIR=/usr/WS1/topopt/libs/lido/toss_4_x86_64_ib/2024_02_16_16_37_47/linux-rhel8-ivybridge/clang-14.0.6/petsc-3.20.4-iwtouyzn3b2zdjpofsc6n637ae52lsbr
  PETSC_ARCH=arch-linux2-c-debug do not specify a valid PETSc installation
Call Stack (most recent call first):
  smith/serac/mfem/CMakeLists.txt:254 (find_package)


CMake Error at /usr/tce/backend/installations/linux-rhel8-x86_64/gcc-10.3.1/cmake-3.23.1-mdfqd2l7c33zg7xcvqizwz25vqmp7jfw/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  PETSc could not be found.  Be sure to set PETSC_DIR and PETSC_ARCH.
  (missing: PETSC_INCLUDES PETSC_EXECUTABLE_RUNS)
Call Stack (most recent call first):
  /usr/tce/backend/installations/linux-rhel8-x86_64/gcc-10.3.1/cmake-3.23.1-mdfqd2l7c33zg7xcvqizwz25vqmp7jfw/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  smith/serac/mfem/config/cmake/modules/FindPETSc.cmake:349 (find_package_handle_standard_args)
  smith/serac/mfem/CMakeLists.txt:254 (find_package)
```

### PR DESCRIPTION
**Pull request by chapman39**
This PR will allow usage of PETSc in Serac builds and installs through MFEM or pure PETSc.
* Updates PETSc smoke test and cmake module
* Build TPLs with PETSc Fixes #1070

NOTE: `MFEM_USE_PETSC` is disabled on codevelop due to the following issue: #1082

### CODE REVIEW

**Comment by chapman39** [comment]
I tested the installation by creating a mini cmake project and calling `find_package(Serac)`.

File: cmake/serac-config.cmake.in [hunk 1]
```diff
@@ -139,7 +141,8 @@ if(NOT SERAC_FOUND)
 
   # Petsc
   if(SERAC_USE_PETSC)
-    # Note: Targets not currently imported
+    # Use the installed FindPETSc.cmake
+    find_dependency(PETSc)
```
Comment (white238) [inline cmake/serac-config.cmake.in]: Can you just call include(FindPetsc.cmake) here?

File: cmake/thirdparty/FindPETSc.cmake [hunk 2]
```diff
@@ -1,147 +1,27 @@
...
-blt_register_library(
-    NAME          PETSc
-    INCLUDES      ${PETSC_INCLUDE_DIRS} ${PETSC_CC_INCLUDES}
-    LIBRARIES     ${PETSC_LIB} ${PETSC_EXTERNAL_LIB_BASIC}
-    TREAT_INCLUDES_AS_SYSTEM ON)
+pkg_search_module(PETSC REQUIRED IMPORTED_TARGET PETSc)
 
-# SERAC EDIT END
+# Serac installation seraches for PETSc_FOUND with a lowercase "c" to verify
+set(PETSc_FOUND ${PETSC_FOUND})
```
Comment (white238) [inline cmake/thirdparty/FindPETSc.cmake:64]: Much nicer!

File: src/tests/CMakeLists.txt [hunk 3]
```diff
@@ -30,9 +30,3 @@ endif()
 
 serac_add_tests( SOURCES       axom_smoketest.cpp
                  DEPENDS_ON    axom::inlet)
```
Comment (white238) [inline src/tests/CMakeLists.txt:32]: Why was this moved?

Comment (chapman39) [reply src/tests/CMakeLists.txt:32]: Since it depends on `serac_infrastructure`, i thought it made more sense to move the test to `src/serac/infrastructure/tests`

File: cmake/serac-config.cmake.in (hunk 1 above)
Comment (chapman39) [reply cmake/serac-config.cmake.in]: pretty sure. ill change that.

Comment (chapman39) [reply cmake/serac-config.cmake.in]: changed to:

```cmake
include("${CMAKE_MODULE_PATH}/FindPETSc.cmake")
```

File: src/tests/CMakeLists.txt (hunk 3 above)
Comment (white238) [reply src/tests/CMakeLists.txt:32]: Ah, I didn't look at the source file. It actually only depends on `axom` for the logging (which could easily be removed). I recommend we move it back and swap the `serac_infrastructure` for `axom::slic`.

TPL smoke tests should where it was originally.

Comment (chapman39) [reply src/tests/CMakeLists.txt:32]: ended up needing to include `axom::inlet` but its been moved

File: cmake/thirdparty/FindPETSc.cmake (hunk 2 above)
Comment (chapman39) [reply cmake/thirdparty/FindPETSc.cmake:64]: spoke too soon, lol.

**Comment by jamiebramwell** [comment]
@white238 , can you take a look at this?

**Review by jamiebramwell — APPROVED** [review]
Thanks @chapman39 !

File: cmake/serac-config.cmake.in [hunk 4]
```diff
@@ -22,6 +22,8 @@ if(NOT SERAC_FOUND)
   set(SERAC_INSTALL_PREFIX "@SERAC_INSTALL_PREFIX@")
   set(SERAC_INCLUDE_DIRS "${SERAC_INSTALL_PREFIX}/include")
 
+  list(APPEND CMAKE_MODULE_PATH "${SERAC_INSTALL_PREFIX}/@SERAC_INSTALL_CMAKE_MODULE_DIR@")
+
```
Comment (white238) [inline cmake/serac-config.cmake.in:26]:
```suggestion
```

No longer needed with the direct include below.

File: cmake/serac-config.cmake.in [hunk 5]
```diff
@@ -139,7 +141,7 @@ if(NOT SERAC_FOUND)
 
   # Petsc
   if(SERAC_USE_PETSC)
-    # Note: Targets not currently imported
+    include("${CMAKE_MODULE_PATH}/FindPETSc.cmake")
```
Comment (white238) [inline cmake/serac-config.cmake.in:144]:
```suggestion
    include(FindPETSc.cmake)
```

File: cmake/thirdparty/SetupSeracThirdParty.cmake [hunk 6]
```diff
@@ -482,19 +484,6 @@ if (NOT SERAC_THIRD_PARTY_LIBRARIES_FOUND)
         set(ENABLE_FORTRAN ON CACHE BOOL "" FORCE)
     endif()
```
Comment (white238) [inline cmake/thirdparty/SetupSeracThirdParty.cmake:491]: Apparently we needed to find PETsc several times lol.... Good catch @chapman39 !

**Review by white238 — APPROVED** [review]
Other than the minor suggestions this looks great. Please merge when its ready!

File: cmake/serac-config.cmake.in (hunk 5 above)
Comment (chapman39) [reply cmake/serac-config.cmake.in:144]: When i don't specify the full path it's unable to find the module

```
CMake Error at /usr/workspace/meemee/serac/repo/install1/lib/cmake/serac-config.cmake:166 (include):
  include could not find requested file:

    FindPETSc.cmake
Call Stack (most recent call first):
  CMakeLists.txt:8 (find_package)
```

Comment (white238) [reply cmake/serac-config.cmake.in:144]: Try using `CMAKE_CURRENT_LIST_DIR` then. `CMAKE_MODULE_PATH` is a list of directories. It just happened to be a single entry this time but normally isnt.
